### DISCUSSION TITLE
Incorporate Plasma into the library?

### DISCUSSION THREAD

**Discussion by jrwrigh** [qtile/qtile#4487] [General]
Thinking about this comment: https://github.com/qtile/qtile/discussions/4484#discussioncomment-7137320
> new layouts are absolutely welcome and we're happy to include them where it makes sense to do so.

**How would the community/maintainers feel about integrating [qtile-plasma](https://github.com/numirias/qtile-plasma) into the library?**

## Why?
Currently, the project is [unmaintained](https://github.com/numirias/qtile-plasma/issues/30). @numirias hasn't "officially" set the package as unmaintained, but:
-  there hasn't been an update in 2 years even though there have been major breakages
-  @numirias's Github account has been inactive for about a year
- their twitter has also been silent for over 2 years
- I'm not aware of any successful attempts to contact in him since Feb 2021 (https://github.com/numirias/qtile-plasma/issues/19)
- It's currently broken for modern Python. And PRs to fix it have been waiting for awhile

## Just fork it?
Yeah, that's definitely an option. And one [I kinda volunteered for](https://github.com/numirias/qtile-plasma/issues/30#issuecomment-1601990515). But I haven't had the time (nor do I have the knowledge for actually maintaining it with proper testing).

Also, I think this layout is well used enough to possibly warrant inclusion into qtile. Personally, when I was first looking at switching from i3 to qtile, I was disappointed that qtile didn't have an i3-esque layout available. It wasn't until I found plasma that I decided that switching over to qtile would work well for me. I'm sure I'm not the only one who had this experience.

---
Maybe I'm jumping the gun here on suggesting taking the project over, but figured I'd float the idea since the layout is quite popular (makes it easier for people moving from i3) and currently it's broken (with PR fixes in waiting).

**Comment by elParaguayo** [comment]
The main problem I have with this is that it's a fairly complex piece of code. Adding it to the main repo requires the maintainers to get familiar with the code which I don't really have time for at the moment.

However, it is a good layout and would be nice to include in the main repo if we can.

**Comment by m-col** [comment]
-1 to having it in qtile. It's great code, and was(/is?) popular, and that is really fantastic. I want to encourage third party projects that integrate with qtile because qtile really is a community project and it is designed to be this configurable and hackable. Adding to the main monolith would be in the opposite direction. I think the ideal would be somebody else picking it up, taking ownership (whether that's directly of the original repo, or a fork), and maintaining it as a third party project :)
